aptos-move/e2e-tests/src/executor.rs

```diff
@@ -736,40 +823,98 @@ impl<O: OutputLogger> FakeExecutorImpl<O> {
-        sequential: bool,
         state_view: &(impl StateView + Sync),
+        config: BlockExecutorConfig,
+        metadata: Option<TransactionSliceMetadata>,
     ) -> Result<Vec<TransactionOutput>, VMStatus> {
-        let config = BlockExecutorConfig {
-            local: BlockExecutorLocalConfig {
-                blockstm_v2: false,
-                concurrency_level: if sequential {
-                    1
-                } else {
-                    usize::min(4, num_cpus::get())
-                },
-                allow_fallback: self.allow_block_executor_fallback,
-                discard_failed_blocks: false,
-                module_cache_config: BlockExecutorModuleCacheLocalConfig::default(),
-            },
-            onchain: onchain_config,
-        };
         let txn_provider = DefaultTxnProvider::new_without_info(txn_block);
-        AptosVMBlockExecutorWrapper::execute_block_on_thread_pool::<
-            _,
-            NoOpTransactionCommitHook<VMStatus>,
-            _,
-        >(
-            self.executor_thread_pool.clone(),
-            &txn_provider,
-            &state_view,
-            // Do not use shared module caches in tests.
-            &AptosModuleCacheManager::new(),
-            config,
-            TransactionSliceMetadata::unknown(),
-            None,
-        )
-        .map(BlockOutput::into_transaction_outputs_forced)
+        let requested = txn_provider.num_txns();
+        let result = {
+            AptosVMBlockExecutorWrapper::execute_block_on_thread_pool::<
+                _,
+                NoOpTransactionCommitHook<VMStatus>,
+                _,
+            >(
+                self.executor_thread_pool.clone(),
+                &txn_provider,
+                &state_view,
+                self.module_cache_manager_opt()
+                    .unwrap_or(&AptosModuleCacheManager::new()),
+                config,
+                metadata.unwrap_or_else(TransactionSliceMetadata::unknown),
+                None,
+            )
+            .map(BlockOutput::into_transaction_outputs_forced)
+        };
+        let mut outputs = result?;
+        // Trim any appended BlockEpilogue output to preserve empty metadata behavior.
+        if outputs.len() > requested {
+            outputs.truncate(requested);
+        }
+        Ok(outputs)
+    }
+
+    /// Returns a reference to the shared module cache manager if enabled (fuzzing/test). Otherwise
+    /// returns [None].
+    fn module_cache_manager_opt(&self) -> Option<&AptosModuleCacheManager> {
+        match &self.block_state {
+            BlockState::Fuzzing(shared) => Some(&shared.manager),
+            BlockState::None => None,
+        }
+    }
+
+    /// Generates a [TransactionSliceMetadata::Block] when running with a shared cache (fuzzing/test)
+    /// to enable cache reuse across calls. For normal runs, returns [None].
+    fn get_txn_slice_metadata(&self, _mode: ExecutorMode) -> Option<TransactionSliceMetadata> {
```

Comment:
nit: instead of option, you can return `TransactionSliceMetadata::unknown` for `BlockState::None`? so there is less matching